File: fixtures/scale/python/large/cipherlab/checks/check_orbitstream.py
from cipherlab.ciphers.orbitstream import orbitstream_encrypt, orbitstream_decrypt
from cipherlab.specs.orbitstream_spec import orbitstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_orbitstream():
    spec = orbitstream_spec()
    for text in corpus_stream():
        sealed = orbitstream_encrypt(text, spec["key"])
        opened = orbitstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_orbitstream.rb
require_relative 'orbitstream_encrypt'
require_relative 'orbitstream_decrypt'
require_relative 'corpus_stream'

def check_orbitstream
  corpus_stream.each do |text|
    sealed = orbitstream_encrypt(text, 12)
    opened = orbitstream_decrypt(sealed, 12)
    return false if opened != text
  end
  true
end
